### PR TITLE
Update ESLint to v6

### LINKED ISSUE #326: Add eslint 6 to CI
Do we test both eslint@5 and eslint@6 in CI like https://travis-ci.org/yannickcr/eslint-plugin-react/builds/548497708. Or do we drop eslint@5 support?

### PR DESCRIPTION
**Pull request by fisker**
Fixes #326

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you change `peerDependencies` too?

**Comment by fisker** [comment]
not done yet

**Comment by fisker** [comment]
I'm not able to fix `no-abusive-eslint-disable` test, should we fake a plugin or ?

**Comment by fisker** [comment]
And please review test and source changes, hope I'm not making any mistake

**Comment by sindresorhus** [comment]
> I'm not able to fix no-abusive-eslint-disable test, should we fake a plugin or ?

@TiagoDanin Any suggestions?

**Comment by TiagoDanin** [comment]
Or replace invalid rules (name) to valid rules. Create fake rules also work.
Example:
`eval(); // eslint-disable-line plugin/rule` ~> `throw new Error(); // eslint-disable-line unicorn/error-message`

The import is test if the [regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/c2b8e76fb20f6ec087d3338225a626836d5e3810/rules/no-abusive-eslint-disable.js#L4) works correctly with different type of definitions (`rule`, `@scope/rule`, `@scope/plugin/rule` and `plugin/rule`).

**Comment by fisker** [comment]
@TiagoDanin @sindresorhus 

I found a way to define non-existing rule

File: rules/prefer-exponentiation-operator.js [hunk 1]
```diff
@@ -22,10 +22,8 @@ const parseArgument = (source, arg) => {
 			return text;
 		case 'CallExpression':
 			return text;
-		case 'UnaryExpression':
-			return text;
 		default:
-			// Handle cases like Math.pow(2, 2-1);
+			// Handle cases like Math.pow(-0, 2-1);
 			return `(${text})`;
 	}
```
Comment (sindresorhus) [inline rules/prefer-exponentiation-operator.js]: @MrHen You might want to review this part.

**Comment by sindresorhus** [comment]
@fisker Great! Thanks fixing this.

**Pull request merged**